metrics/metrics_cli.go

```diff
@@ -0,0 +1,186 @@
+		cli.createSummary(count, mc)
+	case "all":
+		cli.createAll(count, mc, isRandom)
+	default:
+		log.Println("Invalid type")
+	}
+	log.Println("Serving on address: " + address)
+	if isRandom {
+		log.Println("Producing randomized metrics per type")
+
+	} else {
+		log.Println("Producing " + fmt.Sprintf("%d", count) + " metric(s) per type")
+	}
+	log.Println("Updating at a frequency of " + fmt.Sprintf("%d", mc.interval) + " seconds")
+	http.HandleFunc("/", healthCheckHandler)
+	http.Handle("/metrics", promhttp.HandlerFor(promRegistry, promhttp.HandlerOpts{}))
+
+	log.Fatal(http.ListenAndServe(address, nil))
+}
+
+func (cli *CommandLine) createCounter(count int, mc metricCollector) {
+	mc.registerCounter(count)
+	go mc.updateCounter()
+}
+
+func (cli *CommandLine) createGauge(count int, mc metricCollector) {
+	mc.registerGauge(count)
+	go mc.updateGauge()
+}
+
+func (cli *CommandLine) createHistogram(count int, mc metricCollector) {
+	mc.registerHistogram(count)
+	go mc.updateHistogram()
+
+}
+
+func (cli *CommandLine) createSummary(count int, mc metricCollector) {
+	mc.registerSummary(count)
+	go mc.updateSummary()
+}
+
+// createAll generates all 4 metric types
+// If isRandom is sent as true, createAll will generate randomized metrics. Otherwise createALl will steadily create the 4 types of metrics with a fixed count (provided by the user
+func (cli *CommandLine) createAll(count int, mc metricCollector, isRandom bool) {
+
+	if isRandom {
+		idx := rand.Intn(4)
+		lower := 1
+		upper := 4
+		amount := rand.Intn(upper-lower) + lower
+		metrics := []string{"counter", "gauge", "histogram", "summary"}
+		rands := []int{rand.Intn(200), rand.Intn(200), rand.Intn(200), rand.Intn(200)}
+		max := rands[0]
+		for i := 0; i < len(rands); i++ {
+			if rands[i] > max {
+				max = rands[i]
+			}
+		}
+		for i := 0; i <= amount; i++ {
+			if idx >= len(metrics) {
+				idx = 0
+			}
+			str := metrics[idx]
+			idx++
+			switch str {
+			case "counter":
+				cli.createCounter(rands[0], mc)
+			case "gauge":
+				cli.createGauge(rands[1], mc)
+			case "histogram":
+				cli.createHistogram(rands[2], mc)
+			case "summary":
+				cli.createSummary(rands[3], mc)
+			}
+		}
+
+	} else {
+		mc.registerCounter(count)
+		mc.registerGauge(count)
+		mc.registerHistogram(count)
+		mc.registerSummary(count)
+		go mc.updateMetrics(count)
+
+	}
+
+}
+
+// Run reads the config file and uses the data as default arguments.
+// These arguments can be overriden by CLI input (see README)
+func (cli *CommandLine) Run() {
+	data, err := ioutil.ReadFile("config.yaml")
+	if err != nil {
+		log.Fatal(err)
+	}
+	var conf Config
+	if err := conf.Parse(data); err != nil {
+		log.Fatal(err)
+	}
+	generateCmd := flag.NewFlagSet("generate", flag.ExitOnError)
+
+	if conf.Type != "" {
+		defaultType = conf.Type
+	}
+	if conf.Count <= 0 {
+		defaultCount = conf.Count
+	}
+	if conf.Frequency <= 0 {
+		defaultFreq = conf.Frequency
+	}
+	if !conf.Random {
+		defaultRand = conf.Random
+	}
```

Comment:
This doesn't feel right.  I'd expect the defaults to be constants and not changed by configuration, merely superseded.

I would suggest looking at [spf13/cobra](https://github.com/spf13/cobra) and [spf13/viper](https://github.com/spf13/viper) for command structure and configuration handling, respectively.  [This article](https://carolynvanslyck.com/blog/2020/08/sting-of-the-viper/) provides a good description of how to combine the two to achieve the following precedence of data sources for configuration values:

```
CLI flags > environment variables > config file > defaults
```